### PR TITLE
Separating Out Access Logging 

### PR DESCRIPTION
**Pull request by noah-paige**
### Feature or Bugfix

- Refactoring

### Detail
-  Move access logging to a separate environment logging bucket (rather than env default bucket used for athena queries and profiling job artifacts)

### Relates
- <URL or Ticket>

### Security
Please answer the questions below briefly where applicable, or write `N/A`. Based on
[OWASP 10](https://owasp.org/Top10/en/).

- Does this PR introduce or modify any input fields or queries - this includes
fetching data from storage outside the application (e.g. a database, an S3 bucket)?
  - Is the input sanitized?
  - What precautions are you taking before deserializing the data you consume?
  - Is injection prevented by parametrizing queries?
  - Have you ensured no `eval` or similar functions are used?
- Does this PR introduce any functionality or component that requires authorization?
  - How have you ensured it respects the existing AuthN/AuthZ mechanisms?
  - Are you logging failed auth attempts?
- Are you using or adding any cryptographic features?
  - Do you use a standard proven implementations?
  - Are the used keys controlled by the customer? Where are they stored?
- Are you introducing any new policies/roles/users?
  - Have you used the least-privilege principle? How?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by noah-paige** [comment]
Tested:

- [ ] CICD Pipeline
- [ ] Migration Script Upgrade
- [ ] Environment Stack Update
- [ ] Dataset Stack Update
- [ ] Migration Script Downgrade

File: backend/dataall/core/environment/cdk/environment_stack.py [hunk 1]
```diff
@@ -165,30 +165,44 @@ def __init__(self, scope, id, target_uri: str = None, **kwargs):
...
             removal_policy=RemovalPolicy.RETAIN,
             block_public_access=s3.BlockPublicAccess.BLOCK_ALL,
             versioned=True,
             enforce_ssl=True,
         )
-        default_environment_bucket.policy.apply_removal_policy(RemovalPolicy.RETAIN)
-        self.default_environment_bucket = default_environment_bucket
-
-        default_environment_bucket.add_to_resource_policy(
+        default_environment_log_bucket.policy.apply_removal_policy(RemovalPolicy.RETAIN)
```
Comment (dlpzx) [inline backend/dataall/core/environment/cdk/environment_stack.py:179]: Isn't this already define in line 174?

File: backend/dataall/core/environment/cdk/environment_stack.py [hunk 2]
```diff
@@ -165,30 +165,44 @@ def __init__(self, scope, id, target_uri: str = None, **kwargs):
...
+            bucket_name=self._environment.EnvironmentDefaultBucketName,
+            encryption=s3.BucketEncryption.S3_MANAGED,
+            removal_policy=RemovalPolicy.RETAIN,
+            block_public_access=s3.BlockPublicAccess.BLOCK_ALL,
+            versioned=True,
+            enforce_ssl=True,
+            server_access_logs_bucket=default_environment_log_bucket,
+            server_access_logs_prefix=f'access_logs/{self._environment.EnvironmentDefaultBucketName}/',
+        )
+        default_environment_bucket.policy.apply_removal_policy(RemovalPolicy.RETAIN)
```
Comment (dlpzx) [inline backend/dataall/core/environment/cdk/environment_stack.py:203]: same as above, isn't the removal policy already defined in line 196

File: backend/dataall/core/environment/api/types.py [hunk 3]
```diff
@@ -100,6 +100,7 @@
         gql.Field('subscriptionsConsumersTopicName', type=gql.String),
         gql.Field('subscriptionsProducersTopicName', type=gql.String),
         gql.Field('EnvironmentDefaultBucketName', type=gql.String),
+        gql.Field('EnvironmentLogsBucketName', type=gql.String),
```
[the remark this answers is not in the document]
Comment (dlpzx) [reply backend/dataall/core/environment/api/types.py:103]: Although I see it is used in the integration tests

**Review by dlpzx — APPROVED** [review]
Left some nit comments

File: backend/dataall/core/environment/cdk/environment_stack.py (hunk 1 above)
Comment (noah-paige) [reply backend/dataall/core/environment/cdk/environment_stack.py:179]: removal policy applies to the bucket only - the assocaited policy would get deleted unless we include this line 
```
        default_environment_log_bucket.policy.apply_removal_policy(RemovalPolicy.RETAIN)
```

which then potentially exposes our bucket

File: backend/dataall/core/environment/api/types.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (noah-paige) [reply backend/dataall/core/environment/api/types.py:103]: it is used in testings

could theoretically be used elsewhere but currentically not used in our front end client as you mentioned

File: backend/dataall/core/environment/cdk/environment_stack.py (hunk 2 above)
Comment (noah-paige) [reply backend/dataall/core/environment/cdk/environment_stack.py:203]: same response - need to explicitly retain associated resources of bucket (i.e. bucket policy)

**Pull request merged**